client/src/ui/components/models/buildings/worldmap/Structures.tsx

```diff
@@ -0,0 +1,88 @@
+import { useDojo } from "@/hooks/context/DojoContext";
+import { useEntityQuery } from "@dojoengine/react";
+import { Has, getComponentValue } from "@dojoengine/recs";
+import { useGLTF } from "@react-three/drei";
+import { useEffect, useMemo } from "react";
+import * as THREE from "three";
+import { getUIPositionFromColRow } from "@/ui/utils/utils";
+import { StructureType } from "@bibliothecadao/eternum";
+import useUIStore from "@/hooks/store/useUIStore";
+
+export const Structures = () => {
+  const models = useMemo(
+    () => [
+      useGLTF("/models/buildings/castle.glb"),
+      useGLTF("/models/buildings/castle.glb"),
+      useGLTF("/models/buildings/hyperstructure.glb"),
+      useGLTF("/models/buildings/bank.glb"),
+      useGLTF("/models/buildings/mine.glb"),
+      useGLTF("/models/buildings/castle.glb"),
+    ],
+    [],
+  );
+
+  const { setup } = useDojo();
+
+  const existingStructures = useUIStore((state) => state.existingStructures);
+  const setExistingStructures = useUIStore((state) => state.setExistingStructures);
+
+  const builtStructures = useEntityQuery([Has(setup.components.Structure)]);
+
+  const _tmp = useMemo(() => {
+    return builtStructures.map((entity) => {
+      const position = getComponentValue(setup.components.Position, entity);
+      const structure = getComponentValue(setup.components.Structure, entity);
+      const type = StructureType[structure!.category as keyof typeof StructureType];
+      return {
+        col: position!.x,
+        row: position!.y,
+        type: type as StructureType,
+        entity: entity,
+      };
+    });
+  }, [builtStructures.length]);
+
+  useEffect(() => {
```

Comment:
this could be a useMemo probably with recs query inside. Might be better to move this into hook aswel.

Reply:
agreed